lisa/notifiers/json.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from decimal import Decimal
+from enum import Enum
+from typing import Any, List, Type, cast
+
+from dataclasses_json import dataclass_json
+
+from lisa import constants, messages, notifier, schema
+
+
+@dataclass_json()
+@dataclass
+class JsonSchema(schema.Notifier):
+    path: str = "lisa.json"
+
+
+class Json(notifier.Notifier):
+    """
+    The Json notifier is used to output the perf test results in JSON format.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "json"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return JsonSchema
+
+    def _received_message(self, message: messages.MessageBase) -> None:
+        if isinstance(message, messages.PerfMessage):
+            message_dict = {}
+            for key, value in message.__dict__.items():
+                if isinstance(value, Enum):
+                    value = value.value
+                elif isinstance(value, datetime) or isinstance(value, Decimal):
+                    value = str(value)
+                message_dict[key] = value
+
+            if not os.path.exists(self._report_path):
+                with open(self._report_path, "w", encoding="utf-8") as json_file:
```

Comment:
Don't open a file multiple times, it's low efficiency. You can open it when initializing, and close it in finalizing.

Reply:
updated